detect_secrets/filters/heuristic.py

```diff
@@ -113,3 +113,53 @@ def is_non_text_file(filename: str) -> bool:
         '.zip',
     ),
 )
+
+
+def is_templated_secret(secret: str) -> bool:
+    """
+    Filters secrets that are shaped like: {secret}, <secret>, or ${secret}.
+    """
+    if secret[0] == '$':
+        return True
+
+    if (
+        (secret[0] == '{' and secret[-1] == '}')
+        or (secret[0] == '<' and secret[-1] == '>')
+        or (secret[0] == '$' and secret[1] == '{' and secret[-1] == '}')
```

Comment:
Is it possible for a secret to be 1 character and `IndexError`ing on this line?

Reply:
It's a fair point, but like super unlikely. Entropy scanners definitely require more than one character to trip it up, and regexes should require more than one character to meet the bar. After all, a one-character secret is a terrible secret.

I'll add a try/except clause for any keyword filter trip ups.